packages/suite/src/components/suite/bluetooth/BluetoothConnect.tsx

```diff
@@ -0,0 +1,199 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import {
+    bluetoothActions,
+    prepareSelectAllDevices,
+    selectAdapterStatus,
+    selectKnownDevices,
+    selectNearbyDevices,
+} from '@suite-common/bluetooth';
+import { TimerId } from '@trezor/type-utils';
+
+import { BluetoothLoading } from './BluetoothLoading';
+import { BluetoothPairingPin } from './BluetoothPairingPin';
+import { BluetoothScanningList } from './BluetoothScanningList';
+import { BluetoothSelectedDevice } from './BluetoothSelectedDevice';
+import { BluetoothDeniedForSuite } from './errors/BluetoothDeniedForSuite';
+import { BluetoothNotEnabled } from './errors/BluetoothNotEnabled';
+import { BluetoothVersionNotCompatible } from './errors/BluetoothVersionNotCompatible';
+import { DesktopBluetoothDevice } from '../../../actions/bluetooth/DesktopBluetoothDevice';
+import { bluetoothConnectDeviceThunk } from '../../../actions/bluetooth/bluetoothConnectDeviceThunk';
+import { bluetoothDisconnectDeviceThunk } from '../../../actions/bluetooth/bluetoothDisconnectDeviceThunk';
+import { bluetoothStartScanningThunk } from '../../../actions/bluetooth/bluetoothStartScanningThunk';
+import { bluetoothStopScanningThunk } from '../../../actions/bluetooth/bluetoothStopScanningThunk';
+import { closeModalApp } from '../../../actions/suite/routerActions';
+import { useDispatch, useSelector } from '../../../hooks/suite';
+
+const SCAN_TIMEOUT = 30_000;
+const UNPAIRED_DEVICES_LAST_UPDATED_LIMIT_SECONDS = 30;
+
+type BluetoothConnectProps = {
+    uiMode: 'spatial' | 'card';
+};
+
+const selectAllDevices = prepareSelectAllDevices<DesktopBluetoothDevice>();
+
+export const BluetoothConnect = ({ uiMode }: BluetoothConnectProps) => {
+    const dispatch = useDispatch();
+
+    const [waitingForFirstUpdate, setWaitingForFirstUpdate] = useState<boolean>(true);
+    const [selectedDeviceId, setSelectedDeviceId] = useState<string | null>(null);
+    const [scannerTimerId, setScannerTimerId] = useState<TimerId | null>(null);
+
+    const bluetoothAdapterStatus = useSelector(selectAdapterStatus);
+    const allDevices = useSelector(selectAllDevices);
+    const knownDevices = useSelector(selectKnownDevices);
+    const nearbyDevices = useSelector(selectNearbyDevices);
+
+    const lasUpdatedBoundaryTimestamp =
+        Date.now() / 1000 - UNPAIRED_DEVICES_LAST_UPDATED_LIMIT_SECONDS;
+
+    const devices = allDevices.filter(it => {
+        const isDeviceUnresponsiveForTooLong =
+            it.lastUpdatedTimestamp < lasUpdatedBoundaryTimestamp;
+
+        if (isDeviceUnresponsiveForTooLong) {
+            // If the device is connected or paired (it may have been paired in the OS system directly)
+            // => do not filter it based isDeviceUnresponsiveForTooLong
+
+            return (
+                it.connected ||
+                it.paired ||
+                knownDevices.find(knownDevice => knownDevice.id === it.id) !== undefined
```

Comment:
My favourite nit suggestion
```suggestion
                knownDevices.some(knownDevice => knownDevice.id === it.id)
```